src/components/Card.js

```diff
@@ -3,72 +3,128 @@
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteSubmitPopup,
+    handleCardLike,
+    checkLikeStatus,
+  ) {
+    this._cardData = cardData;
+    
+    this._name = cardData.name;
+    this._link = cardData.link;
+    this._cardId = cardData._id;
+    this._isLiked = cardData.isLiked;
     this._cardSelector = cardSelector;
-    //this._cardElement = this._getCardElement();
     this._handleImageClick = handleImageClick;
+    this._handleDeleteSubmitPopup = handleDeleteSubmitPopup;
+    this.handleCardLike = handleCardLike;
+    this.checkLikeStatus = checkLikeStatus;
+    
+    
   }
 
-  _setEventListeners() {
-    // this._likeButton.addEventListener("click", () => {
-    //   this._handleLikeIcon();
-    // });
+  getId() {
+    return this._cardId;
+  }
+  
+  toggleLikeIcon(cardData) {
+    
+    if (this._likeButton.classList.contains("cards__like-button_active")) {
+      this._likeButton.classList.remove("cards__like-button_active");
+      console.log(cardData, "is TOGGLED TO unliked");
+    } else {
+      this._likeButton.classList.add("cards__like-button_active");
+      console.log(cardData, "is TOGGLED TO liked");  
+    }}
 
-    // this._deleteButton.addEventListener("click", () => {
-    //   this._handleDeleteCard();
-    // });
+    
+    
+    submitButtonLoadingState(isLoading) {
+      
+      if (this._submitButton) {
+        this._submitButton.textContent = isLoading
+        ? this._defaultButtonText
+        : "Saving...";
+      }
+    }
+    
+    getView() {
+      this._element = document
+        .querySelector(this._cardSelector)
+        .content.querySelector(".cards__list-item")
+        .cloneNode(true);
+  
+      this._likeButton = this._element.querySelector(".cards__like-button");
+  
+      this._deleteButton = this._element.querySelector(".cards__delete-button");
+      // this._confirmDeleteButton = this._element.querySelector(
+      //   "#confirm-delete-button"
+      // );
+  
+      this._cardTitleElement = this._element.querySelector(".cards__title");
+      this._cardImageElement = this._element.querySelector(".cards__image");
+  
+      this._cardTitleElement.textContent = this._name;
+      this._cardImageElement.src = this._link;
+      this._cardImageElement.alt = this._name;
+  
+      this.checkLikeStatus(this._isLiked, this._cardData);
+      
+      this._setEventListeners();
+      
+      return this._element;
+    }
+  
+    _setEventListeners() {
+      this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteSubmitPopup(this._cardId, this._element);
+        console.log(this._cardId, "to delete?");
+      });
+    
+      this._likeButton.addEventListener("click", () => { 
+        this.handleCardLike(this._isLiked, this._cardData, this._cardId);    
+       this.toggleLikeIcon(this._isLiked);
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.